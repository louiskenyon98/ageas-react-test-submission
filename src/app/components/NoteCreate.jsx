import React from 'react';
import {connect} from 'react-redux';
import {postNote} from '../actions';
// import NoteForm from './NoteForm';
import NoteFormContainer from '../containers/notes/NoteForm';
import styles from '../../themes/style.scss';


class NoteCreate extends React.Component {

    onSubmit = (formValues) => {
        this.props.postNote(formValues);
    };
    render() {
        return (
            <div className={`col-xl-4 offset-xl-4 col-lg-6 offset-lg-3 col-sm-8 offset-sm-2 ${styles["note-creator"]}`}>
                <h2>Create New Note</h2>
                {/*<NoteForm onSubmit={this.onSubmit} />*/}
                <NoteFormContainer
                    onSubmit={this.onSubmit}
                />
            </div>

        )
    }
}

export default connect(null, {postNote})(NoteCreate);
