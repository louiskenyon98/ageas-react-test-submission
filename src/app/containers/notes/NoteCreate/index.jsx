import React from 'react';
import {connect} from 'react-redux';
import {getNotes, postNote} from '../../../actions/note/notes.action';
import NoteFormContainer from '../NoteForm';
import styles from '../NotesContainer.styles.scss';


export class NoteCreateContainer extends React.Component {
    constructor(props) {
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
    }

    onSubmit(formValues) {
        this.props.postNote(formValues, this.props.getNotes);
    };

    render() {
        return (
            <div className={`col-xl-4 offset-xl-4 col-lg-6 offset-lg-3 col-sm-8 offset-sm-2 ${styles.noteCreator}`}>
                <h2>Create New Note</h2>
                <NoteFormContainer
                    onSubmit={this.onSubmit}
                />
            </div>

        )
    }
}

export default connect(
    null,
    {postNote, getNotes}
)(NoteCreateContainer);
