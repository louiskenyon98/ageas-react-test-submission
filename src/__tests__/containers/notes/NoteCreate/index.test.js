import React from 'react';
import {shallow} from 'enzyme';
import {NoteCreateContainer} from '../../../../app/containers/notes/NoteCreate/index';
import NoteFormContainer from '../../../../app/containers/notes/NoteForm/index';

describe('NoteCreateContainer', () => {
    let props = {};
    let wrapper = () => shallow(<NoteCreateContainer {...props}/>);
    beforeEach(() => {
        props = {
            postNote: jest.fn(),
            getNotes: jest.fn()
        }
    });
    describe('render', () => {
        it('should render NoteCreateContainer with Redux Form as child', () => {
            expect(wrapper()).toMatchSnapshot();
        })
    });
    describe('functionality', () => {
        it('should call postNote onSubmit', () => {
            const formValues = {
                hello: 123
            };
            wrapper().find(NoteFormContainer).first().props().onSubmit(formValues);
            expect(props.postNote).toHaveBeenCalledTimes(1);
            expect(props.postNote).toHaveBeenCalledWith(formValues, expect.anything());
        })
    })
});
