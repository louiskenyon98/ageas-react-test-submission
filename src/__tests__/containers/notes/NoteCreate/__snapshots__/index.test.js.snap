// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCreateContainer render should render NoteCreateContainer with Redux Form as child 1`] = `
<div
  className="col-xl-4 offset-xl-4 col-lg-6 offset-lg-3 col-sm-8 offset-sm-2 note-creator"
>
  <h2>
    Create New Note
  </h2>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={true}
    forceUnregisterOnUnmount={false}
    form="noteForm"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    onSubmit={[Function]}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
    validate={[Function]}
  />
</div>
`;
